### PR TITLE
Fix api user vulnerability

### LINKED ISSUE #2267: Fix api schema vulnerability
# Description
Based on #2266, but the quick version.
User should never be nested in another object, because we want to be extra careful about auth permissions on the User object. Remove it from Pool and PoolCandidate, and any other types containing it. If that breaks any existing queries in the app (rerun codegen and then build typescript to find out) then replace the User type in those objects with other, less detailed versions, like Applicant or UserPublicProfile.

# Acceptance Criteria
- [ ] the full User type is never nested in another object

### PR DESCRIPTION
**Pull request by tristan-orourke**
Removes User as a nested object from other types. In places, User has been replaced with another type with a subset of User fields.

Its still possible to create and update users when creating or updating Pool Candidates, but those mutations were renamed to AsAdmin to be really explicit about that not being a normal ability.

Resolves #2267

### CODE REVIEW

**Review by petertgiles — CHANGES_REQUESTED** [review]
I think there's some trouble on the pool candidate forms.  When I created the candidate with the _No Longer Interested_ status they were created with the _Available Status_.  When I ran an update on the same record I got  some errors:
image
Of course it's possible these pre-existed... hopefully e2e testing starts soon!

I noticed that search requests are created without a pool assigned but I guess that's the lack of auth on talent again.

**Review by JamesHuf — CHANGES_REQUESTED** [review]
I'm getting an error anytime I try to update a user currently. I think this is new. 
image

**Comment by JamesHuf** [comment]
> @JamesHuf That probably has to do with the issue being addressed by @petertgiles in this PR #2330

Yeah you're probably right, I'll update my review.

**Review by JamesHuf — APPROVED** [review]
As Vachan pointed out my issue is not actually a part of this PR.

**Comment by tristan-orourke** [comment]
I've approved and merge #2330, I'll merge it into this branch so the issues aren't confused

**Comment by tristan-orourke** [comment]
@petertgiles the Status not being saved was a bug in the CreatePoolCandidate component, it just wasn't being submitted and it defaults to AVAILABLE. Fixed in https://github.com/GCTC-NTGC/gc-digital-talent/pull/2331/commits/155192dc8bb3f3f129baaa5e85819cfde0c72676

File: api/graphql/schema.graphql [hunk 1]
```diff
@@ -724,9 +720,14 @@ input OperationalRequirementBelongsToMany {
...
-input CreatePoolCandidateInput {
+input ConnectOrCreateBelongsTo {
+    connect: ID
+    create: CreateUserInput
+}
+
+input CreatePoolCandidateAsAdminInput {
     pool: PoolBelongsTo!
-    user: UserBelongsTo!
+    user: ConnectOrCreateBelongsTo! # TODO: This should be switched to UserBelongsTo as soon as possible. Permission to create a PoolCandidate shouldn't imply permission to create a User.
```
[the remark this answers is not in the document]
Comment (tristan-orourke) [reply api/graphql/schema.graphql]: Good suggestion. I created an issue and referenced it in comment - I'll expand on the issue later.

**Comment by petertgiles** [comment]
The create pool candidate now has right status.  An update is still failing though with the _missing primary key_ error.  I bet it's the user update part of the mutation since it doesn't have an user ID 
image

**Comment by tristan-orourke** [comment]
It turns out that using the `update` [nested mutation](https://lighthouse-php.com/master/eloquent/nested-mutations.html#belongsto) requires you to include the id - and it can be a different id! Meaning that it would allow you to update any arbitrary User, while attaching that User to the PoolCandidate. That is unintended behaviour, so I've removed it entirely.
On the form side, I've simply disabled the User field inputs. They're still there for read-only convenience. The form submits now.

**Review by petertgiles — APPROVED** [review]
Great solution!  Thanks for all the tidying up on this one too!

**Pull request merged**